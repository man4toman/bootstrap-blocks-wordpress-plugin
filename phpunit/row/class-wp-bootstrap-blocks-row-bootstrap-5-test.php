<?php
/**
 * Row tests.
 *
 * @package wp-bootstrap-blocks
 */

/**
 * Class WP_Bootstrap_Blocks_Row_Test
 */
class WP_Bootstrap_Blocks_Row_Bootstrap_5_Test extends WP_Bootstrap_Blocks_UnitTestCase {
	/**
	 * Name of current block.
	 *
	 * @var string $blockname
	 */
	protected $blockname = 'row';

	/**
	 * Runs before each test
	 */
	public function setUp() {
		parent::setUp();
		// Enable Bootstrap 5
		update_option( \WP_Bootstrap_Blocks\Settings::OPTION_PREFIX . 'bootstrap_version', 5 );
	}

	/**
	 * Runs after each test
	 */
	public function tearDown() {
		parent::tearDown();
		delete_option( \WP_Bootstrap_Blocks\Settings::OPTION_PREFIX . 'bootstrap_version' );
	}

	/**
	 * Tests row no gutters.
	 */
	public function test_row_no_gutters() {
		$variant = 'no-gutters-bootstrap-5';
		list( $expected, $actual ) = $this->get_block_output( $variant );
		$this->assertEquals( $expected, $actual );
	}

	/**
	 * Tests row horizontal gutters.
	 */
	public function test_row_horizontal_gutters() {
		$variant = 'horizontal-gutters';
		list( $expected, $actual ) = $this->get_block_output( $variant );
		$this->assertEquals( $expected, $actual );
	}

	/**
	 * Tests row vertical gutters.
	 */
	public function test_row_vertical_gutters() {
		$variant = 'vertical-gutters';
		list( $expected, $actual ) = $this->get_block_output( $variant );
		$this->assertEquals( $expected, $actual );
	}
}
