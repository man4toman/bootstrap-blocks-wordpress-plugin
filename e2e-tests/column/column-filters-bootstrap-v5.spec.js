/**
 * WordPress dependencies
 */
import {
	activatePlugin,
	deactivatePlugin,
	createNewPost,
	getEditedPostContent,
} from '@wordpress/e2e-test-utils';
import { insertRowBlock } from '../row/row-helper';
import { selectColumnBlock } from './column-helper';
import {
	getCheckboxValueByLabel,
	getDataValuesOfElement,
	getInputValueByLabel,
	getSelectedValueBySelectLabel,
	openSidebarPanelWithTitle,
	selectOption,
	selectOptionIsAvailable,
} from '../helper';

describe( 'column block filters', () => {
	beforeAll( async () => {
		await activatePlugin( 'wp-bootstrap-blocks-test-bootstrap-v5' );
		await activatePlugin( 'wp-bootstrap-blocks-test-column-filters' );
	} );

	afterAll( async () => {
		await deactivatePlugin( 'wp-bootstrap-blocks-test-bootstrap-v5' );
		await deactivatePlugin( 'wp-bootstrap-blocks-test-column-filters' );
	} );

	beforeEach( async () => {
		await createNewPost();
	} );

	it( 'wp_bootstrap_blocks_column_default_attributes should override default attributes with Bootstrap 5', async () => {
		await insertRowBlock();
		await selectColumnBlock();

		// Xxl Columm size value should be set
		await openSidebarPanelWithTitle( 'Column size' );
		expect( await getInputValueByLabel( 'Xxl Column count' ) ).toMatch(
			'5'
		);

		// Check if attributes are set correctly
		expect( await getEditedPostContent() ).toMatchSnapshot();
	} );
} );
