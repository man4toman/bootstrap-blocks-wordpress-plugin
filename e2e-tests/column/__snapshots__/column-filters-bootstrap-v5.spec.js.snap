// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`column block filters Bootstrap 5 wp_bootstrap_blocks_column_default_attributes should override default attributes with Bootstrap 5 1`] = `
"<!-- wp:wp-bootstrap-blocks/row -->
<!-- wp:wp-bootstrap-blocks/column {\\"sizeMd\\":6} /-->

<!-- wp:wp-bootstrap-blocks/column {\\"sizeMd\\":6} /-->
<!-- /wp:wp-bootstrap-blocks/row -->"
`;
